sktime/performance_metrics/base/_base.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+__author__ = ["Ryan Kuhns"]
+__all__ = ["BaseMetric"]
+
+import inspect
+from sklearn.base import BaseEstimator
+
+
+class BaseMetric(BaseEstimator):
+    """Base class for defining metrics in sktime. Extends
+    scikit-learn's BaseEstimator.
+    """
+
+    def __init__(self, func, name=None):
+        self._func = func
+        self.name = name if name is not None else func.__name__
+
+    def __call__(self, y_true, y_pred, **kwargs):
+        NotImplementedError("abstract method")
+
+    @classmethod
+    def _all_tags(cls):
```

Comment:
Perhaps we should refactor our current `BaseEstimator` in `sktime.base._base` into 
* a `BaseObject` inheriting from scikit-learn's `BaseEstimator` for handling of hyper-parameters and tags, 
* a separate `BaseEstimator` inheriting from `BaseObject` for estimators (classifiers, forecasters, etc) which additionally has the `is_fitted` and `check_is_fitted` functionality.

@RNKuhns @fkiraly any thoughts on this? If we do this, this should go into a separate PR. Happy to leave it as it is in this one.

Reply:
But perhaps add a comment that `_all_tags` is copied from `BaseEstimator`

Reply:
Yes, I've always thought that the parameter/tag functionality in `BaseEstimator` should be split off into a separate package that implements "parametric classes" :-) 
So I'm onboard with the proposed factoring - *but* what does this have to do with the metrics module? This sounds more like an independent refactor effort.

Reply:
It's related to the metrics because metrics now inherit from `BaseEstimator` which equips them with hyper-parameter and tag functionality which we want but also functionality for managing states (fitted vs unfitted) which we don't want.

Reply:
I'm for it. I can add an issue about this later.  As part of the PR to close that, we can pretty easily switch the `BaseMetric` to inherit from the new `BaseObject`.